src/lib/routes/client-api/feature.ts

```diff
@@ -64,12 +66,13 @@ export default class FeatureController extends Controller {
         config: IUnleashConfig,
     ) {
         super(config);
-        const { clientFeatureCaching } = config;
+        const { clientFeatureCaching, flagResolver } = config;
         this.featureToggleServiceV2 = featureToggleServiceV2;
         this.segmentService = segmentService;
         this.clientSpecService = clientSpecService;
         this.openApiService = openApiService;
         this.logger = config.getLogger('client-api/feature.js');
+        this.seenEtags = new Map<string, string>();
```

Comment:
This makes me afraid of filling up the memory with garbage. Maybe we can have a fixed-size data structure discarding the oldest values.

I'm not sure if this is a valid concern, but I do expect the key to change frequently over time.

Reply:
Ivar is experimenting with etag calculation based on query params (req.query and key joined together) and the most recent eventId instead of the full body of the response. That experiment will supersede this PR. So we might be closing this.